File: kotlin-react-router-dom-legacy/karakum-legacy/remix-run-router/karakum/inheritanceModifiers/modifyInterfaceInheritance.js
import ts from "typescript"

export default (node) => {
    if (
        ts.isInterfaceDeclaration(node)
        && (node.name.text === "AgnosticRouteMatch"
            || node.name.text === "AgnosticPatchRoutesOnNavigationFunction")
    ) {
        return ""
    }

    if (
        ts.isInterfaceDeclaration(node)
        || ts.isTypeAliasDeclaration(node)
        || ts.isTypeLiteralNode(node)
        || ts.isIntersectionTypeNode(node)
    ) {
        return "sealed"
    }

    return null
}

File: kotlin-typescript/karakum/src/jsMain/kotlin/wrappersgenerator/typescript/inheritanceModifiers/modifyInterfaceInheritance.kt
package wrappersgenerator.typescript.inheritanceModifiers

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.InheritanceModifierContext
import typescript.*

fun modifyInterfaceInheritance(node: Node, context: InheritanceModifierContext) = nullable {
    ensure(
        isInterfaceDeclaration(node)
                || isTypeAliasDeclaration(node)
                || isTypeLiteralNode(node)
                || isIntersectionTypeNode(node)
    )

    "sealed"
}
